cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -385,10 +386,7 @@ template <typename It>
 _CCCL_HOST_DEVICE auto make_iterator_kernel_arg(It it) -> kernel_arg<It>
 {
   kernel_arg<It> arg;
-  // since we switch the active member of the union, we must use placement new or construct_at. This also uses the copy
-  // constructor of It, which works in more cases than assignment (e.g. thrust::transform_iterator with
-  // non-copy-assignable functor, e.g. in merge sort tests)
-  ::cuda::std::__construct_at(&arg.iterator, it);
+  arg.iterator = it;
```

Comment:
In a chat with @bernhardmgruber it came up that there have been improvements to thrust iterators since this comment was originally written, which make it copyable. @miscco I think we discussed that it would be good to have your blessing on this change.

Reply:
@shwina could you please remind me again why this change here is necessary? And if @miscco does not have a better idea, we should just coment out `construct_at` and add a comment that this would be the correct code but we have to do it differently because <your rational>.

Reply:
IIRC, it's because when calling from `c.parallel`, `it` is a pointer, not an iterator value and so the call to `__construct_at` fails.

Reply:
If this is the case then we have a bug in libcu++, since it works in `std`: https://godbolt.org/z/Gxhfeaz89

Reply:
My bad! After I introduced https://github.com/NVIDIA/cccl/blob/2491a9115ec4c8cc104cebd0fba582ea82ac86a1/cub/cub/device/dispatch/dispatch_transform.cuh#L85-L86, `c.parallel` doesn't end up calling this function. 

I've undone this change. Thanks for catching it!